.ado/templates/prep-and-pack-nuget.yml

```diff
@@ -15,11 +16,38 @@ parameters:
   signMicrosoft: false
 
 steps:
-  - task: DownloadPipelineArtifact@2
-    displayName: 'Download ReactWindows Artifact'
+  - ${{ if eq(length(parameters.slices), 0) }}:
+    - powershell: |
+        Write-Host "##vso[task.setvariable variable=basePlatform]x64"
+        Write-Host "##vso[task.setvariable variable=baseConfiguration]Release"
+      displayName: Set baseConfiguration and basePlatform to x64.Release
+  - ${{ else }}:
+    - powershell: |
+        $slices = "${{ join(',', parameters.slices) }}".split(',')
+        $base = $slices[0].split('.')
+        Write-Host "##vso[task.setvariable variable=basePlatform]$($base[0])"
+        Write-Host "##vso[task.setvariable variable=baseConfiguration]$($base[1])"
+      displayName: Set baseConfiguration and basePlatform
+      
+  - ${{ each slice in parameters.slices }}:
+    - task: DownloadPipelineArtifact@2
+      displayName: 'Download ${{ parameters.artifactName }}.${{ slice }} Artifact'
+      inputs:
+        artifact: ${{ parameters.artifactName }}.${{ slice }}
+        path: $(System.DefaultWorkingDirectory)/ReactWindows
```

Comment:
does the published artifact have the right subfolder structure? Or does the path here need configuration/platform?

Reply:
I guess it works, maybe a future cleanup item we can file a workitem for to have a cleaner artifacts.. by extracting the folders to avoid a potential double-write case in downloading multiple artifacts to the same folder...

Reply:
Yeah, ideally we would have the artifact be the folder inside of $(platform)/$(configuration), then have the download step know where to place it based on the slice being downloaded. I ended up not doing that mostly because grabbing those values from the slice list in the download step is itself a little bit janky.

I originally had slices as an array of object of the form `{configuration: string, platform: string}`, but that form could not be serialized to be passed in its entirety to other scripts, like we do for stripping platforms from the NuSpec. So for now I pass everything around as a string array and try to avoid grabbing props from it except where needed. This means needing to dive into powershell or the like if we want to do anything semantically with the slice though.

Reply:
Figured out a creative way to serialize typed slices I think. Will move to those, so we can do this part.

Reply:
Better yet, I missed that you can convert them to JSON. https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#converttojson

Reply:
Fixed. Slices are now typed objects so that we can use platform/configuration in paths. I couldn't find it documented anywhere, but subscripts of array parameters seem to work correctly as well.